### PR TITLE
Choose better path dynamically to download hadoop

### PR DESCRIPTION
**Pull request by sneaxiy**
Downloading hadoop is quite slow in China. Try to choose the download URL dynamically.

### CODE REVIEW

File: docker/ci/install-hadoop.bash [hunk 1]
```diff
@@ -16,9 +16,14 @@
 set -e
 
 # NOTE: require external exported HADOOP_VERSION.
-HADOOP_SITE="https://archive.apache.org/dist/hadoop/common"
-axel --quiet --output /tmp/hadoop.tar.gz \
-     $HADOOP_SITE/hadoop-"$HADOOP_VERSION"/hadoop-"$HADOOP_VERSION".tar.gz
+
+HADOOP_DYN_SITE="https://www.apache.org/dyn/closer.cgi/hadoop/common/hadoop-$HADOOP_VERSION/hadoop-$HADOOP_VERSION.tar.gz"
+
+hadoop_url=`curl -s $HADOOP_DYN_SITE | grep -o -E "href\=\"https:\/\/.*\/hadoop-$HADOOP_VERSION.tar.gz\"" | head -n 1 | awk -F'"' '{print $2}'`
```
Comment (wangkuiyi) [inline docker/ci/install-hadoop.bash]: Need a comment to explain this line.

Also, use `\` to make this multiple lines, each less than 80 characters please.

File: docker/ci/install-hadoop.bash [hunk 2]
```diff
@@ -16,9 +16,14 @@
...
+
+echo "Download hadoop from $hadoop_url"
```
Comment (wangkuiyi) [inline docker/ci/install-hadoop.bash]: hadoop => Hadoop

Add `...` after the message.

File: docker/ci/install-hadoop.bash [hunk 3]
```diff
@@ -16,9 +16,14 @@
...
+
+axel --quiet --output /tmp/hadoop.tar.gz $hadoop_url
```
Comment (wangkuiyi) [inline docker/ci/install-hadoop.bash]: Why HADDOP_DYN_SITE is capitalized, but haddop_url is not? Should we use the same convention of naming variables?

File: docker/ci/install-hadoop.bash (hunk 1 above)
Comment (sneaxiy) [reply docker/ci/install-hadoop.bash]: Done.

File: docker/ci/install-hadoop.bash (hunk 2 above)
Comment (sneaxiy) [reply docker/ci/install-hadoop.bash]: Done.

File: docker/ci/install-hadoop.bash (hunk 3 above)
Comment (sneaxiy) [reply docker/ci/install-hadoop.bash]: Done.

**Review by wangkuiyi — APPROVED** [review]

**Pull request merged**